text_file_to_ui/text_file_to_ui/Form1.cs

```diff
@@ -18,9 +18,35 @@ private void btnLoadFile_Click_1(object sender, EventArgs e)
             UserDisplayer userDisplayer = new UserDisplayer();
             _ = new User();
             Reader reader = new Reader();
-            User a = userParser.UserParse(reader);
-
-            if (a != null)
+            try
+            {
+                User a = userParser.UserParse(reader);
+                if (a != null)
+                {
+                    txtFullName.Text = userDisplayer.NameFull(a);
+                    txtYearOfBirth.Text = a.YearOfBirth.ToString();
+                    txtCity.Text = a.City.ToString();
+                    txtFaculty.Text = a.Faculty.ToString();
+                    txtRole.Text = a.Role.ToString();
+                    lblTimesOpenedNum.Text = a.TimesRead.ToString();
+                    if (a.Role == "student")
+                    {
+                        txtRoleDep.Text = a.FavoriteCourse.ToString();
+                        lblRoleDep.Text = "Favorite course:";
+                    }
+                    else
+                    {
+                        txtRoleDep.Text = a.Department.ToString();
+                        lblRoleDep.Text = "Katedra:";
+                    }
+                }
+            }
```

Comment:
Od svega označenoga, samo je metoda `UserParse` opasna za izbacivanje exceptiona. Sve ostalo ne može puknuti.
Stoga, bespotrebno opterećujemo `try` blok i kôd, smanjena je čitljivost zbog jednog taba više itd.

Moj savjet: neka UserParse metoda try-catcha exception. Ako je bio exception (datoteka nije otvorena ili što već), u redu, neka `UserParse` vrati 'null'. A forma, ona ako je dobila 'null', ispisuje error-MessageBox, a ako je dobila objekt korisnika najnormalnije odrađuje sav kôd i ispisuje korisnikove podatke.